### PR TITLE
Fix uploadlimits checks

### PR DESCRIPTION
**Pull request by HLeithner**
Pull Request for Issue #35853

### Summary of Changes
check if upload, post and memory limit is set to unlimited before checking required minimum limits

### Testing Instructions
Install extensions
Upload file in media manger.
if possible with different values set in for upload_max_filesize, post_max_size, memory_limit in php.ini

### Actual result BEFORE applying this Pull Request
doesn't work if any of the 3 vars is <= 0

### Expected result AFTER applying this Pull Request
works

### CODE REVIEW

File: administrator/components/com_installer/src/Model/WarningsModel.php [hunk 1]
```diff
@@ -98,76 +98,98 @@ public function getItems()
...
-			// 16MB
-			$messages[] = array('message' => Text::_('COM_INSTALLER_MSG_WARNINGS_LOWMEMORYWARN'),
-					'description' => Text::_('COM_INSTALLER_MSG_WARNINGS_LOWMEMORYDESC'));
-		}
-		elseif ($memory_limit < ($minMemory * 1.5) && $memory_limit != -1)
-		{
-			// 24MB
-			$messages[] = array('message' => Text::_('COM_INSTALLER_MSG_WARNINGS_MEDMEMORYWARN'),
-					'description' => Text::_('COM_INSTALLER_MSG_WARNINGS_MEDMEMORYDESC'));
+		if ($memory_limit > -1) {
```
Comment (brianteeman) [inline administrator/components/com_installer/src/Model/WarningsModel.php]:
```suggestion
		if ($memory_limit > -1)
		{
```

**Comment by PhilETaylor** [comment]
Ive said it before. Repeatedly. memory_limit doesn't effect uploading of files! Basic PHP. 

https://github.com/joomla/joomla-cms/issues/35360#issuecomment-907127303
https://github.com/joomla/joomla-cms/pull/35392#issuecomment-907135264

**Comment by HLeithner** [comment]
> Ive said it before. Repeatedly. memory_limit doesn't effect uploading of files! Basic PHP.

I would say the same but there are 2 "problems"

> [#35360 (comment)](https://github.com/joomla/joomla-cms/issues/35360#issuecomment-907127303) 

First it's mentioned in the documentation https://php.itronic.at/manual/en/ini.core.php#ini.post-max-size

```Generally speaking, memory_limit should be larger than post_max_size.```

> [#35392 (comment)](https://github.com/joomla/joomla-cms/pull/35392#issuecomment-907135264)

If you look at the mediamanager then you will see that it loads the images as base64 encoded string...

sure both things doesn't effect the upload directly but is a failsafe for later usage. Does you see a negative impact?

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on ab4610a28de09ecfe00f02a43f056bbc061c51d8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36023">issues.joomla.org/tracker/joomla-cms/36023</a>.</sub>

**Comment by PhilETaylor** [comment]
I have tested this item :white_check_mark: successfully on ab4610a28de09ecfe00f02a43f056bbc061c51d8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36023">issues.joomla.org/tracker/joomla-cms/36023</a>.</sub>

**Comment by Fedik** [comment]
R2C<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36023">issues.joomla.org/tracker/joomla-cms/36023</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**